### DISCUSSION TITLE
error in creating new case of coupled atm, land and ocean

### DISCUSSION THREAD

**Discussion by gourikadam** [E3SM-Project/E3SM#6737] [E3SM model help]
Dear Team,

I wish to create new case of coupled atm, land and ocean. So I took compset  F20TR_MMF1, in which I got mane errors which I could resolved after installing submodules (https://github.com/E3SM-Project/E3SM/discussions/6699#top)
but e3sm.bldlog files shows error
1. "/scratch/um_cdac2/met_models/CDACP_Int/E3SM/components/mpas-framework/src/external/ezxml/ezxml.c(647): error: identifier "MADV_SEQUENTIAL" is undefined
          madvise(m, l, MADV_SEQUENTIAL); // optimize for sequential access"
2. tmp/ifortDCFjDB.i90: catastrophic error: **Internal compiler error: segmentation violation signal raised** Please report this error along with the circumstances in which it occurred in a Software Problem Report.  Note: File and line given may not be explicit cause of this error

Since , I could not make ice component stub in previous case and could not resolve the errors, i tried with other compset "F20TR_SCREAM1"
 Here also I got catastrophic error. Our HPC team asked me to change cmake version and with it i got following errors

1. with cmake_3.27.4
gcc 7.5.0

Error message:
'cmake: /home/apps/gcc-7.5.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by cmake)
cmake: /home/apps/gcc-7.5.0/lib64/libstdc++.so.6: version `CXXABI_1.3.13' not found (required by cmake)
cmake: /home/apps/gcc-7.5.0/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by cmake)' from dir '/scratch/um_cdac2/met_models/CDACP_Int/e3sm/e3sm_scratch/atm_F20TR_SCREAM1/cmaketmp'

2. with cmake_3.27.4
gcc 11.1.0

Error message:
/home/apps/gcc-11.1.0/include/c++/11.1.0/tuple(649): error: pack "_UElements" does not have the same number of elements as "_Elements"
            __and_<is_nothrow_constructible<_Elements, _UElements>...>::value

3. with cmake 3.27.9 ( loaded using spack)
gcc 11.1.0

Error message:
i. Kokkos_NumericTraits.hpp(38): error: expression must have a constant value
  template <> struct infinity_helper<float> { static constexpr float value = HUGE_VALF; };
ii. /home/apps/gcc-11.1.0/include/c++/11.1.0/tuple(649): error: pack "_UElements" does not have the same number of elements as "_Elements"
            __and_<is_nothrow_constructible<_Elements, _UElements>...>::value;

4. with cmake 3.27.9 ( loaded using spack)
gcc 7.5.0

Error message:
/tmp/ifortp0Vf8Q.i90: catastrophic error: **Internal compiler error: segmentation violation signal raised** Please report this error along with the circumstances in which it occurred in a Software Problem Report.

Not clearly understanding where I went wrong

Regards,
Gouri

**Comment by rljacob** [comment]
gcc 7.5 is definitely to old.  Add it looks like 11.1 can't handle the C++ features in Kokkos.  Try 12.1 which we do test with.

**Reply by rljacob** [reply]
Or 11.2.
